frontend/src/app/shared/components/interests-dropdown/interests-dropdown.component.ts

```diff
@@ -8,23 +8,23 @@ import { getTags } from '@shared/utils/tagsForInterests';
     styleUrls: ['./interests-dropdown.component.sass'],
 })
 export class InterestsDropdownComponent {
-    toggle: boolean = false;
+    toggle = false;
 
     @Input() inputList: IIcon[] = getTags();
 
-    @Output() selectedInterests = new EventEmitter<string[]>();
+    @Output() selectedInterests = new EventEmitter<Set<string>>();
 
-    outputList: string[] = [];
+    outputList = new Set<string>();
 
     selectInterest($event: Event) {
         const ev = $event.target as HTMLInputElement;
         const numb = parseInt(ev.id, 10);
         const { checked } = ev;
 
         if (checked) {
-            this.outputList = this.outputList.concat(this.inputList[numb].icon_name);
+            this.outputList.add(this.inputList[numb].icon_name);
```

Comment:
@tatianahutii recommended collections to be immutable. Why did you change it?

Reply:
avoid using mutations